app/hotels/src/allHotels/searchForm/SearchForm.js

```diff
@@ -108,24 +75,26 @@ export default class SearchForm extends React.Component<Props, State> {
         <View style={styles.row}>
           <DatePicker
             placeholder="Start date"
-            format={DATE_FORMAT}
-            date={this.state.checkin}
+            format={DISPLAY_DATE_FORMAT}
+            date={moment(search.checkin, DATA_DATE_FORMAT).toDate()}
```

Comment:
I think `DatePicker` should accept `Date` and do necessary formatting on its own. This interface is hard to (re)use. Also - the underlying `react-native-datepicker` accepts `Date` objects so we can improve it this way...